### PR TITLE
Issue #309, path and element ranges

### PR DESCRIPTION
**Pull request by grechaw**
JSON envelopes #308
Aesthetic fixes in  #304
Also, #303 also is fixed in this commit

### CODE REVIEW

File: entity-services/src/test/java/com/marklogic/entityservices/tests/TestInstanceConverterGenerator.java [hunk 1]
```diff
@@ -221,10 +221,10 @@ public void testConversionModuleExtractions() throws TestEvalException, JsonProc
...
             Document controlDom = builder.parse(this.getClass().getResourceAsStream(controlFilePath));
 
-            // logger.debug("Control doc");
-            // debugOutput(controlDom);
-            // logger.debug("Actual doc wrapped");
-            // debugOutput(actualInstance);
+             logger.debug("Control doc");
+             debugOutput(controlDom);
+             logger.debug("Actual doc wrapped");
+             debugOutput(actualInstance);
```
Comment (bsrikan) [inline entity-services/src/test/java/com/marklogic/entityservices/tests/TestInstanceConverterGenerator.java]: Do you want to keep this logger and debugger uncommented?

Comment (grechaw) [inline entity-services/src/test/java/com/marklogic/entityservices/tests/TestInstanceConverterGenerator.java]: no, I need to comment that out again, good catch. tests are rather more verbose with these uncommented :)

**Comment by grechaw** [comment]
OK I've pushed those comments as a new commit on this PR

File: entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy [hunk 2]
```diff
@@ -304,13 +420,68 @@ declare function {$prefix}:instance-to-envelope(
...
     }}
 }};
 
 
+(:
+ : @param $entity-instance an instance, as returned by an extract-instance
+ : function
+ : @return A document which wraps both the canonical instance and source docs.
+ :)
+declare function {$prefix}:instance-to-envelope(
```
Comment (bsrikan) [inline entity-services/src/main/xdmp/entity-services/entity-services-codegen.xqy]: Do we need this wrapper, now that we have explicit xml and json envelope functions? How or where can this be of use?

File: entity-services/src/main/xdmp/entity-services/entity-services.xqy [hunk 3]
```diff
@@ -334,16 +344,10 @@ declare function es:init-instance(
  :)
 declare function es:add-attachments(
     $instance as json:object,
-    $source-node as item()*,
```
Comment (bsrikan) [inline entity-services/src/main/xdmp/entity-services/entity-services.xqy]: Why did we remove input $source-node?

**Comment by bsrikan** [comment]
There are couple of unit test failures when I ran test from branch json-envelopes @grechaw 1> verifySchemaValidation() and testModelXMLConersion() . 
That and other than the questions I have posted, I am good and think we can merge this PR so I can start absorbing the changes in functional tests and write new tests for json envelope implementation.

**Comment by grechaw** [comment]
OK I'll merge this then.  schema work is progressing nicely actually.

**Pull request merged**